### PR TITLE
332 fixes

### LINKED ISSUE #304: Make map items (circles) active/clickable
At the moment ony the name below map items are clickable to make item information appear. If the user clicks an icon/circle, nothing happens, although this might be the expected behaviour.

- [ ] copy behaviour of the item label to the item circle, so that information appear on item click, too

### PR DESCRIPTION
**Pull request by bonndan**
* Fixes that SVG is loaded twice (cleanup of previous #332 fix attempt)
* closes #304 (small UX improvement)

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java [hunk 1]
```diff
@@ -81,11 +81,11 @@ public DomContent render() {
             circle.attr("stroke-dasharray", 5);
             circle.attr("opacity", 0.7);
         }
-        ContainerTag inner = SvgTagCreator.g(circle, content, children)
-                .attr("class", "hexagon");
+        ContainerTag inner = SvgTagCreator.g(circle, content, children);
 
         return SvgTagCreator.g(inner, icon)
-                .attr("class", "hexagon-group")
```
Comment (mfbieber) [inline src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: Was this what made the hexagons before? Are we not having the hexagons anymore?

Comment (mfbieber) [reply src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: Starting in line 92 in this class there is a TODO and a commented private method. Is this dead code? Do we need it, or can we remove it?

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItemLabel.java [hunk 2]
```diff
@@ -27,11 +27,10 @@ public DomContent render() {
...
 
-        ContainerTag g = SvgTagCreator.g(null, labelText).attr("class", "label");
-        g.attr("id", id);
-        g.attr("data-identifier", identifier);
-        g.attr("text-anchor", "middle");
-        return g;
+        return SvgTagCreator.g(null, labelText)
+                .attr("class", "label")
+                .attr("id", id)
+                .attr("text-anchor", "middle");
```
Comment (mfbieber) [inline src/main/java/de/bonndan/nivio/output/map/svg/SVGItemLabel.java:33]: What does removing the "data-identifier" attribute do?

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java [hunk 3]
```diff
@@ -81,11 +81,11 @@ public DomContent render() {
...
+                .attr("data-identifier", this.id)
+                .attr("class", "item")
```
Comment (mfbieber) [inline src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: What does the data-identifier and the class attribute change here?

File: src/test/java/de/bonndan/nivio/output/map/svg/SVGItemLabelTest.java [hunk 4]
```diff
@@ -21,8 +24,8 @@ public void regression184() {
         foo.setLandscape(landscape);
 
 
-        SVGItemLabel svgItemLabel = new SVGItemLabel(foo);
-        String render = svgItemLabel.render().render();
+        SVGItem svgItem = new SVGItem(null, new LayoutedComponent(foo), new Point2D.Double(0,0));
+        String render = svgItem.render().render();
```
Comment (mfbieber) [inline src/test/java/de/bonndan/nivio/output/map/svg/SVGItemLabelTest.java:28]: How are we still testing the `SVGItemLabel` now?

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItemLabel.java (hunk 2 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/map/svg/SVGItemLabel.java:33]: "data-identifier" is used in javascript to obtain the item identifier. This has been moved to a parent element (see below).

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java (hunk 3 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: The element gets on onClick event (all ".item" elements) and the identifier is used to show correct item data in a dialogue.

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java (hunk 1 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: The item is neither a "hexagon" anymore nor do we need the class here anymore.

The dead code has been copied to https://github.com/dedica-team/nivio/issues/303

File: src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java [hunk 5]
```diff
@@ -3,33 +3,53 @@
...
+     */
+    public LocalIcons(@Value("${nivio.iconFolder:#{null}}") String iconFolder) {
+        if (iconFolder != null) {
+            this.iconFolder = iconFolder.endsWith(File.separator) ? iconFolder : iconFolder + File.separator;
+        } else {
+            this.iconFolder = DEFAULT_ICONS_FOLDER;
+        }
+        defaultIcon = getIconUrl(DEFAULT_ICON.getIcon()).orElseThrow(() -> {
+            throw new RuntimeException(String.format(initErrorMsg, this.iconFolder));
+        });
```
Comment (mfbieber) [inline src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java:52]: Okay, the problem I had was indeed resolved by obtaining the npm dependency. 👍

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -28,6 +28,7 @@ nivio:
   baseUrl: ${NIVIO_BASE_URL:}
   version: #project.version#
   pollingMilliseconds: 30000 # 30 secs
+  # iconFolder: /a/local/path
```
Comment (mfbieber) [inline src/main/resources/application.yml:31]: What do we need this for?

Comment (mfbieber) [reply src/main/resources/application.yml:31]: Ah, okay, to provide it, if it is different than the default folder 💡

File: src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java (hunk 5 above)
Comment (mfbieber) [reply src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java:52]: Why not making the parameter `@Nonnull` and using the `DEFAULT_ICONS_FOLDER` as the fallback value?

File: src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java (hunk 3 above)
Comment (mfbieber) [reply src/main/java/de/bonndan/nivio/output/map/svg/SVGItem.java:88]: Okay, I haven't looked into the frontend logic querying for these yet.

File: src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java (hunk 5 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/icons/LocalIcons.java]: I've had to look up how that is done using injected values, but it works. Change is pushed.

File: src/test/java/de/bonndan/nivio/output/layout/OrganicLayouterTest.java [hunk 7]
```diff
@@ -168,7 +168,7 @@ public void renderLandscapeItemModelWithMagicLabels() throws IOException {
     public void renderCSV() throws IOException {
 
         HttpService httpService = new HttpService();
-        IconService iconService = new IconService(new LocalIcons(null), new VendorIcons(httpService));
```
Comment (mfbieber) [inline src/test/java/de/bonndan/nivio/output/layout/OrganicLayouterTest.java]: This looks pretty now 🦋

**Pull request merged**